compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -690,7 +690,9 @@ impl<'tcx> WfPredicates<'tcx> {
                     // All of the requirements on type parameters
                     // have already been checked for `impl Trait` in
                     // return position. We do need to check type-alias-impl-trait though.
-                    if self.tcx.is_type_alias_impl_trait(def_id) {
+                    if self.tcx.is_impl_trait_in_trait(def_id)
```

Comment:
I'm confused about why do I need this. Need to investigate better.

Reply:
maybe we want ...

```rust
fn is_type_alias_impl_trait<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId) -> bool {
    match tcx.hir().get_if_local(def_id) {
        Some(Node::Item(hir::Item { kind: hir::ItemKind::OpaqueTy(opaque), .. })) => {
            matches!(opaque.origin, hir::OpaqueTyOrigin::TyAlias)
        }
        Some(_) => bug!("tried getting opaque_ty_origin for non-opaque: {:?}", def_id),
        None => false,
    }
}
```

or check `opt_rpitit_info` at the top to avoid hitting the `bug!` call for things that doesn't have HIR.

@compiler-errors thoughts?.

Reply:
Wait, just feed `type_alias_impl_trait` query the value `false`...?